engine/common/rpc/convert/headers.go

```diff
@@ -91,7 +91,53 @@ func MessageToBlockHeader(m *entities.BlockHeader) (*flow.Header, error) {
 			ProposerID:         MessageToIdentifier(m.ProposerId),
 			ChainID:            *chainId,
 			LastViewTC:         lastViewTC,
-		},
+		})
+		if err != nil {
+			return nil, fmt.Errorf("failed to create root header body: %w", err)
+		}
+
+		rootHeader, err := flow.NewRootHeader(flow.UntrustedHeader{
+			HeaderBody:  *rootHeaderBody,
+			PayloadHash: MessageToIdentifier(m.PayloadHash),
+		})
+		if err != nil {
+			return nil, fmt.Errorf("failed to create root header: %w", err)
+		}
+
+		return rootHeader, nil
+	}
+
+	headerBody, err := flow.NewHeaderBody(flow.UntrustedHeaderBody{
+		ParentID:           MessageToIdentifier(m.ParentId),
+		Height:             m.Height,
+		Timestamp:          m.Timestamp.AsTime(),
+		View:               m.View,
+		ParentView:         m.ParentView,
+		ParentVoterIndices: m.ParentVoterIndices,
+		ParentVoterSigData: m.ParentVoterSigData,
+		ProposerID:         MessageToIdentifier(m.ProposerId),
+		ChainID:            *chainId,
+		LastViewTC:         lastViewTC,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("could not build header body: %w", err)
+	}
+	header, err := flow.NewHeader(flow.UntrustedHeader{
+		HeaderBody:  *headerBody,
 		PayloadHash: MessageToIdentifier(m.PayloadHash),
-	}, nil
+	})
+	if err != nil {
+		return nil, fmt.Errorf("could not build header: %w", err)
+	}
+
+	return header, nil
+}
+
+// IsRootBlockHeader reports whether this is a root block header.
+// It returns true only if all of the fields required to build a Header are zero/nil.
+func IsRootBlockHeader(m *entities.BlockHeader) bool {
+	return MessageToIdentifier(m.ParentId) == flow.ZeroID &&
+		m.ParentVoterIndices == nil &&
```

Comment:
```suggestion
	return m.ParentVoterIndices == nil &&
```
Parent ID is non-zero in case of spork root block. 
https://github.com/onflow/flow-go/blob/fd81ed5345e0a699a7961821be27c6e03376fcb6/cmd/bootstrap/run/block.go#L10-L13